systemverilog-plugin/UhdmAst.cc

```diff
@@ -1088,7 +1089,22 @@ AST::AstNode *UhdmAst::process_value(vpiHandle obj_h)
         // Surelog reports constant integers as a unsigned, but by default int is signed
         // so we are treating here UInt in the same way as if they would be Int
         case vpiUIntVal:
+            if (val.value.uint > std::numeric_limits<std::uint32_t>::max()) {
+                // an integer is by default signed, so use 'sd despite the variant vpiUIntVal
+                strValType = "'sd";
+                string str_value = std::to_string(val.value.uint);
+                val.value.str = strdup(str_value.c_str());
+                break;
+            }
+            [[fallthrough]];
         case vpiIntVal: {
+            if (val.value.integer > std::numeric_limits<std::int32_t>::max()) {
+                strValType = "'sd";
+                string str_value = std::to_string(val.value.integer);
```

Comment:
I wonder if this case can be simplified further. Will parsing the strings return incorrect values for smaller constants? Maybe we just need to set the size in that case and then parse them from strings?

Reply:
Parsing decimals as strings can also be used for numbers shorter than 32 bits with success. However, I left this case deliberately to be handled by the older flow which considers Typespec before creating the constant Node. In case when we use strings, we don't actually set any size. This is done by the function which parses the string. In this case, it's `VERILOG_FRONTEND::const2ast`.

I've added new test cases here:
https://github.com/chipsalliance/UHDM-integration-tests/pull/696

Reply:
@rkapuscik  ping, could you take a look at this again? I think it's ready, although I removed the test case exceeding 64 bits, as this PR only covers <32;64> bits and I've run tests without bumping other submodules.
Tests were successful: https://github.com/antmicro/yosys-systemverilog/pull/1196
To my knowledge, there's nothing else I would be expected to change.

Reply:
OK, I think it's acceptable for this limited scope. LGTM.